src/coreclr/jit/lclmorph.cpp

```diff
@@ -1012,18 +1007,16 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
             return IndirTransform::None;
         }
 
-        if (indir->OperIs(GT_IND))
+        if (indir->OperIs(GT_IND)) // IND<struct>
         {
-            // Skip TYP_STRUCT IND nodes, it's not clear what we can do with them.
-            // Normally these should appear only as sources of variable sized copy block
-            // operations (DYN_BLK) so it probably doesn't make much sense to try to
-            // convert these to local nodes.
+            // TODO-ADDR: add this case to the "don't expect" assert above; it requires updating
+            // "cpblk" import to not create such nodes for block copies of known size.
             return IndirTransform::None;
         }
 
-        if ((user == nullptr) || !user->OperIs(GT_ASG, GT_CALL, GT_RETURN))
+        if (!user->OperIs(GT_ASG, GT_CALL, GT_RETURN))
         {
-            // TODO-ADDR: remove unused indirections.
+            // TODO-ADDR: define the contract for "COMMA(..., LCL<struct>)".
```

Comment:
For context, `COMMA(..., LCL_VAR<struct>)` is currently an unsupported IR shape because we fail to mark the local DNER in morph in a number of cases, which is fatal for promoted locals. We also have code in args morphing which just discards `COMMA`s for such tree.